File: sdk/workbuddy-enterprises/python-sdk/src/workbuddy_enterprise/resources/enterprise.py

from __future__ import annotations

from typing import Any

from workbuddy_enterprise.resources._base import Resource
from workbuddy_enterprise.response import ApiResponse


class EnterpriseResource(Resource):
    def get_info(self) -> ApiResponse[dict[str, Any]]:
        return self._as_map(self._get("/info"))

    def get_license(self) -> ApiResponse[dict[str, Any]]:
        return self._as_map(self._get("/license"))

File: sdk/workbuddy-enterprises/rust-sdk/src/resources/enterprise.rs
use crate::client::Client;
use crate::error::Result;
use crate::response::ApiResponse;
use serde_json::Value;

pub struct EnterpriseResource<'a> {
    pub(crate) client: &'a Client,
}

impl EnterpriseResource<'_> {
    pub fn get_info(&self) -> Result<ApiResponse<Value>> {
        self.client.get_json("/info", &[])
    }

    pub fn get_license(&self) -> Result<ApiResponse<Value>> {
        self.client.get_json("/license", &[])
    }
}
